src/providers/info.rs

```diff
@@ -285,6 +287,10 @@ impl<C: Cache> DefaultInfoProvider<'_, C> {
         Ok(result)
     }
 
+    fn generate_token_key(&self) -> String {
+        format!("{}_{}", TOKENS_KEY_BASE, self.chain_id)
+    }
```

Comment:
Should this function be extracted out of `self` and take the `chain_id` as parameter?

This can be used also by `InvalidationPattern.to_pattern_string` – if we change the pattern we need to make sure that it's changed in both places.